vscode/src/chat/clientStateBroadcaster.ts

```diff
@@ -111,31 +112,37 @@ export function startClientStateBroadcaster({
 
     const disposables: vscode.Disposable[] = []
 
+    const sendClientState = debounced(rawSendClientState)
```

Comment:
Do we need a custom `debounce` implementation? We use the one from `lodash` in a couple of places already.

```suggestion
import { debounce } from 'lodash'

    const sendClientState = debounce(rawSendClientState, DEBOUNCE_DELAY)
```

Reply:
Yeah, because it's async and we need to pass in an AbortController. I will add a comment justifying this.

Reply:
I see. Because the debounced function body needs to know when it's canceled too 👍